daisy_workflows/build-publish/rhui/rhua_sync_status.py

```diff
@@ -0,0 +1,154 @@
+  metric = {'type': metric_type}
+  resource_labels = {'project_id': '155767908850',
+                     'location': region,
+                     'namespace': 'rhua-sync-status',
+                     'node_id': mig_name}
+  resource = {'type': 'generic_node',
+              'labels': resource_labels}
+
+  time_series = [{'metric': metric,
+                  'resource': resource,
+                  'points': points}]
+
+  data = {'timeSeries': time_series}
+  print(data)
+  r = requests.post(url, json=data, headers=headers)
+  print(r.text)
+
+
+def PublishRepoAge(seconds_since_update, timestamp, access_token, mig_name):
+  print(seconds_since_update)
+  end_time = time.strftime('%Y-%m-%dT%H:%M:%S-00:00', time.gmtime(timestamp))
+  metric_type = 'custom.googleapis.com/rhua_sync_age'
+
+  points = [{'interval': {'endTime': end_time},
+             'value': {'int64Value': seconds_since_update}}]
+
+  PublishMetric(metric_type, points, access_token, mig_name)
+
+
+def PublishRecentUpdates(recent_updates, access_token, mig_name):
+  if not recent_updates:
+    return
+  metric_type = 'custom.googleapis.com/rhua_updates'
+  points = []
+  for update in recent_updates:
+    end_time = time.strftime('%Y-%m-%dT%H:%M:%S-00:00', time.gmtime(update))
+    point = {'interval': {'endTime': end_time}, 'value': {'boolValue': True}}
+    points.append(point)
+
+  PublishMetric(metric_type, points, access_token, mig_name)
+
+
+def PublishLastHourCount(hour_count, timestamp, access_token, mig_name):
+  print(hour_count)
+  end_time = time.strftime('%Y-%m-%dT%H:%M:%S-00:00', time.gmtime(timestamp))
+  metric_type = 'custom.googleapis.com/rhua_updates_in_hour'
+
+  points = [{'interval': {'endTime': end_time},
+             'value': {'int64Value': hour_count}}]
+
+  PublishMetric(metric_type, points, access_token, mig_name)
+
+
+def GetAccessToken():
+  url = ('http://metadata.google.internal/computeMetadata/v1/instance/'
+         'service-accounts/default/token')
+  headers = {'Metadata-Flavor': 'Google'}
+  r = requests.get(url, headers=headers)
+  access_token = r.json()['access_token']
+  return access_token
+
+
+def GetMIGName():
+  url = ('http://metadata.google.internal/computeMetadata/v1/instance/'
+         'attributes/created-by')
+  headers = {'Metadata-Flavor': 'Google'}
+  r = requests.get(url, headers=headers)
+  full_path = r.text
+  mig_name = full_path.split('/')[-1]
+  return mig_name
+
+
+def GetRegion():
+  url = 'http://metadata.google.internal/computeMetadata/v1/instance/zone'
+  headers = {'Metadata-Flavor': 'Google'}
+  r = requests.get(url, headers=headers)
+  zone = r.text.split('/')[-1]
+  zone_split = zone.split('-')
+  region = '-'.join(zone_split[:-1])
+  return region
+
+
+def main():
+  newest = 0.0
+  now = time.time()
+  hour_ago = now - 3600
+  five_min_ago = now - 300
+  hour_count = 0
+  recent_updates = []
+
+  repodata_list = GetRepodataList(REPODIR)
+
+  for repodata in repodata_list:
```

Comment:
does this block deserve a function?

Reply:
I originally went down that path, but it was confusing since the block needs to return 2 ints and a list. I could return an object, but it didn't seem worth it. I can create an object with this info if you think it would be better.

Reply:
i think you have one function now which finds matching pathnames, then main has to act on them. but what you really want are just times here. if you have the lookup function do the stats, then you're just operating a sorted list of integer time values. then the comparisons you need here could be turned into simple comprehensions.

imagine:
```py3
times = GetRepoDataTimes(REPODIR)
in_last_hour = [x for x in times if x > hour_ago]
in_last_five = [x for x in in_last_hour if x > five_min_ago]
PublishRepoAge(times[0], now, access_token, mig_name)  # It can do the math
PublishLastHourCount(len(in_last_hour), now, access_token, mig_name)
PublishRecentUpdates(in_last_five, access_token, mig_name)
```

Reply:
I like that. I think I'll save it for a future improvement though.